### DISCUSSION TITLE
Perfomance issue with hasPermissionTo() for example

### DISCUSSION THREAD

**Discussion by Sentoki** [spatie/laravel-permission#2064] [Q&A]
Hi everyone! Is it okay, that execution of \Spatie\Permission\Traits\HasPermissions::hasPermissionTo() takes half of time?

Or maybe just something misconfigured?

![image](https://i.imgur.com/vzXKOgH.png)

**Comment by erikn69** [comment]
Hi, which version are you using??
maybe you have a lot of `hasPermissionTo`
If you know a better way, make a PR for fix that

**Comment by PaolaRuby** [comment]
how do you get those execution trees?
